File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
        int maxArea = 0;
        int n = heights.length;

        int start = 0, end = n-1;

        while(start < end){
            int minHt = Math.min(heights[start], heights[end]);
            int width = end - start;
            maxArea = Math.max(maxArea, minHt * width);
            if(heights[start] <= heights[end]){
                start++;
            }
            else{
                end--;
            } 
        }
        return maxArea;
    }
}

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-1.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        n = len(heights)
        maxarea = 0
        for i in range(n):
            ht = 1_000_000_000
            for j in range(i, n): 
                ht = min(ht, heights[j])
                width = j-i+1
                maxarea = max(maxarea, ht*width)
        return maxarea